File: Data Structures & Algorithms/jump-game-ii/submission-0.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        dp = [10000]*len(nums)
        dp[-1] = 0
        n = len(nums)


        for i in range(n - 2, -1, -1):
            end = min(n, i + nums[i] + 1)
            for j in range(i + 1, end):
                dp[i] = min(dp[i], dp[j] + 1)
        
        return dp[0]

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {

        let prefix = new Array(nums.length).fill(1);
        let postfix = new Array(nums.length).fill(1);
        let res = []

        for(let i = 1; i < nums.length; i++){
            prefix[i] = nums[i-1] * prefix[i-1]
        }

        for(let j = nums.length - 2; j > -1; j--){
            postfix[j] = nums[j+1] * postfix[j+1]
        }

        for(let a = 0; a < nums.length; a++){
            res[a] = prefix[a] * postfix[a];
        }

        console.log(prefix, postfix)

        return res;

    }
}
